### PR TITLE
Added symlink dereferencing in fast packaging and tests

### PR DESCRIPTION
**Pull request by vchowdhary**
# TL;DR
Adds a switch to enable dereferencing symlinks during fast packaging in `pyflyte serialize fast workflows`.
cc @wild-endeavor 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added --- added documentation for click switch
 - [ ] Any pending items have an associated Issue

## Complete description
Added a click switch to `pyflyte serialize fast workflows` that enables/disables dereferencing any symlinks encountered during packaging. This only applies to `fast_package`. 

## Tracking Issue
_NA_

No issue associated with this -- this is part of a request from Stripe. Our use case for this is as follows:
Our workflows often import code from utils files that are in sibling directories -- this makes the current `pyflyte package` not quite work for us. Our workaround is to use our bazel build target, which generates a `runfiles` directory that contains all the source code and dependencies, and then serialize from that folder. Some symlinks are generated during the build, so we needed to add `dereference=True` when opening the tar file so that the actual target contents are loaded and downloaded during execution. 

This change should not effect any of the current DataPersistence plugins that are used since they are all remote and would require dereferencing symlinks anyways if they wanted to use the contents. 

## Follow-up issue
_NA_

### CODE REVIEW

**Review by eapolinario — COMMENTED** [review]
This change looks good. Can you also make sure that new flag is added to the other places where `fast_package` is used?

**Comment by vchowdhary** [comment]
> This change looks good. Can you also make sure that new flag is added to the other places where `fast_package` is used?

Thanks for catching that! There was one place that I missed.

**Comment by eapolinario** [comment]
What about https://github.com/flyteorg/flytekit/blob/3137609c6d838a92077b7be27fc92cef78144066/flytekit/tools/repo.py#L98 ?

**Comment by vchowdhary** [comment]
Ah my bad, missed that.

**Review by eapolinario — APPROVED** [review]

**Comment by vchowdhary** [comment]
@eapolinario  
Any idea on when this would be released? We see that there are [monthly releases usually](https://pypi.org/project/flytekit/#history) and one hasn't happened yet for september.
